docs/tutorials/creating-a-runtime-module.md

```diff
@@ -51,184 +73,286 @@ You should be able to successfully compile this template module with:
+and you will get a compile time error.
 
 So when building your module have two options:
 
-**Option 1 (recommended)**
-
-Develop against the `v1.0` branch as the template demonstrates. In this case you will need to update your node template's `Cargo.toml` to depend on Substrate's `branch = 'v1.0'` like the module template currently does, rather than `rev = <git commit hash>`. This will ensure that your module can be used by any node who is also based on the `v1.0` branch
-
-**Option 2**
+**Option 1**
 
-It is also acceptable to develop your module for a specific git commit. In this case you'll need to change the Substrate module template's `Cargo.toml` to use `rev = <git commit hash>` rather than `branch = 'v1.0'`. This means that only nodes which use the exact same git commit hash for their project can use your module.
+If you prefer to have the freshly-baked features from Substrate, you can develop
+your module against a specific git commit of Substrate. In future to upgrade,
+you can update to a newer git commit of Substrate, recompile and test your
+module against the runtime, and then release your module.
 
-If you follow our recommended option 1, your dependency definitions will look like:
+In this case you change the Substrate module `Cargo.toml` to use
+`rev = <git commit hash>`, e.g.:
 
-```toml
-# This dependency is published on crates.io so just use a version
-[dependencies.parity-codec]
-default-features = false
-features = ['derive']
-version = '3.5'
+```TOML
+#--snip--
 
-# These dependencies specify a git branch
 [dependencies.support]
 default_features = false
 git = 'https://github.com/paritytech/substrate.git'
 package = 'srml-support'
-branch = 'v1.0'
+rev = '3dedd246c62255ba6f9b777ecba318dfc2078d85'
+
+#--snip--
+```
+
+This means that only nodes which use the exact same git commit hash for their
+projects can use your module.
+
+**Option 2**
+
+If you prefer to have more stability, you can develop your module against
+Substrate's known release tags (e.g.
+[`v1.0.0`](https://github.com/paritytech/substrate/tree/v1.0.0)). In this case
+you will need to update your node template's `Cargo.toml` to depend on
+Substrate's `tag = 'v1.0.0'`, e.g.:
 
-[dependencies.system]
+```TOML
+#--snip--
+
+[dependencies.support]
 default_features = false
 git = 'https://github.com/paritytech/substrate.git'
-package = 'srml-system'
-branch = 'v1.0'
-# Could also use a specific revision
-# rev = '<commit hash>'
+package = 'srml-support'
+tag = 'v1.0.0'
+
+#--snip--
 ```
 
 ### Your Module's Dev Dependencies
 
-The final section of the `Cargo.toml` file specifies the dev dependencies. These are the dependencies that are needed in your module's tests, but not necessary the actual module itself.
+The final section of the `Cargo.toml` file specifies the dev dependencies.
+These are the dependencies that are needed in your module's tests, but not
+necessary the actual module itself.
 
-
-```toml
+```TOML
+#--snip--
```

Comment:
We should double check if we need a feature flag here or not. I think it was removed in the past, but done so incorrectly.